components/org.wso2.carbon.identity.multi.attribute.login/org.wso2.carbon.identity.multi.attribute.login.resolver.regex/src/main/java/org/wso2/carbon/identity/multi/attribute/login/resolver/regex/RegexResolver.java

```diff
@@ -138,6 +127,15 @@ private void resolveDistinctUsersForClaims(String loginAttribute, List<String> a
             }
         }
 
+        // Check the users from username by default if there are no users found from claims and regex patterns.
+        if (distinctUsers.size() == 0) {
+            if (allowedAttributes.contains(UserCoreClaimConstants.USERNAME_CLAIM_URI)) {
+                List<User> userList = userStoreManager.getUserListWithID(UserCoreClaimConstants.USERNAME_CLAIM_URI,
+                        loginAttribute, null);
```

Comment:
check that the same user is adding to the list

Reply:
Resolved from https://github.com/wso2-extensions/identity-governance/pull/743/commits/5cff3e814be7727922bf3bc29413cc3c71fea72d